sbioapputils/app_runner/yaml_utils.py

```diff
@@ -0,0 +1,313 @@
+                if not parameters[key].get('tooltip'):
+                    parameters[key]['tooltip'] = key
+                if not parameters[key].get('title'):
+                    parameters[key]['title'] = key
+                    
+                # Numeric settings
+                if parameters[key].get('type'):
+                    if parameters[key]['type'] in ['int', 'float']:
+                        input_parameters = _define_from_numeric(parameters[key], key, input_parameters)
+                    
+                    elif parameters[key]['type'] == 'str':
+                        input_parameters, inputs_require_files = _define_from_string(parameters[key], key, input_parameters, inputs_require_files)
+                else:
+                    raise Exception(f"Please define 'type' for parameter {key}")
+ 
+    input_files = _define_files_from_yaml(yaml_dict)
+    
+    workflow_filename = workflow_loc.split('/')[-1]
+    
+    #including workflow name
+    input_parameters.append({
+           "default_value": workflow_filename,
+           "hidden": True,
+           "input_type": "user_input",
+           "name": "workflow_name",
+           "title": "workflow_name",
+           "tooltip": "workflow_name",
+           "type": "text"
+        })
+    
+    settings_config = {
+        "disabledFields": inputs_require_files,
+        "inputsRequireFiles": inputs_require_files,
+        "parameters": {
+            "header": "Set Parameters",
+            "inputs": input_parameters
+        },
+        "uploadOptions": input_files
+    }
+    
+    # Output parameters
+    results_config = {
+        "description": "No description provided"
+    }
+    
+    if yaml_dict.get('output_settings'):
+        if yaml_dict['output_settings'].get('description'):
+            results_config['description'] = yaml_dict['output_settings']['description']
+    
+    app_settings = {
+        "resultsConfig": results_config,
+        "settingsConfig": settings_config
+    }
+            
+    return json.dumps(app_settings)
+
+
+def payload_from_yaml(workflow_loc, config_only = False):
+    """
+    Returns a JSON string of the results_for_payload dictionary and a list of additional artifacts.
+    
+    This can be used to populate a default json file for results
+
+    Parameters:
+    yaml_dict (dict): yaml dict containing output settings.
+
+    Returns:
+    Tuple containing the JSON string of results_for_payload dictionary and the list of additional artifacts.
+    """
+    yaml_dict = get_yaml(workflow_loc)
+    
+    if (yaml_dict['output_settings'].get('folder') and config_only == False):
+        results_for_payload, additional_artifacts = payload_from_folder(yaml_dict['output_settings']['folder'], yaml_dict)
+    else:
+        results_for_payload, additional_artifacts = payload_from_config(yaml_dict)
+    
+    return results_for_payload, additional_artifacts
+    
+
+def _generate_carousel(output_settings_dict, result_type_key):
```

Comment:
why not just accept straight output_settings_dict[result_type_key] in this function?